tests/Tests.AzureAppConfiguration/RefreshTests.cs

```diff
@@ -959,13 +910,32 @@ private Mock<ConfigurationClient> GetMockConfigurationClient()
 
             Response<ConfigurationSetting> GetTestKey(string k, string l, CancellationToken ct)
             {
-                return Response.FromValue(_kvCollection.FirstOrDefault(s => s.Key == k), mockResponse.Object);
+                return Response.FromValue(_kvCollection.FirstOrDefault(s => (s.Key == k && s.Label == l)), mockResponse.Object);
             }
 
+            Response<ConfigurationSetting> GetIfChanged(ConfigurationSetting setting, bool cond, CancellationToken ct)
+            {
+                var newSetting = _kvCollection.FirstOrDefault(s => (s.Key == setting.Key && s.Label == setting.Label));
+                var unchanged = (newSetting.Key == setting.Key && newSetting.Label == setting.Label && newSetting.Value == setting.Value);
+                var response = new MockResponse(unchanged ? 304 : 200);
+                return Response.FromValue(newSetting, response);
+            }
+
+            // We don't actually select KV based on SettingSelector, we just return a deep copy of _kvCollection
             mockClient.Setup(c => c.GetConfigurationSettingsAsync(It.IsAny<SettingSelector>(), It.IsAny<CancellationToken>()))
-                .Returns(new MockAsyncPageable(_kvCollection));
+                .Returns(() =>
+                {
+                    var copy = new List<ConfigurationSetting>();
+                    foreach (var setting in _kvCollection)
+                    {
+                        copy.Add(TestHelpers.CloneSetting(setting));
+                    };
```

Comment:
Here's an alternate way to clone the list.
List<ConfigurationSetting> copy = _kvCollection.Select(setting => TestHelpers.CloneSetting(setting)).ToList();